upgrade-service/docker/Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM sonicdev-microsoft.azurecr.io/debian:bookworm
+
+# Create non-root user for security
+RUN groupadd -r mopd && useradd -r -g mopd -s /bin/false mopd
```

Comment:
Recommend using `adduser --system --group mopd`. This automatically sets the shell an home directory to something that doesn't exist.